File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums.length==1)
        return ;
        int i = 0 ; 
        int j = 0; 
        while(j<nums.length){
            if(i<nums.length){
              if(nums[i]==0){
                    while(j<nums.length&&nums[j]==0){
                        j++ ;
                    }if(j!=nums.length){
                    nums[i] = nums[j] ;
                    nums[j] = 0 ;
                    }

              }
              i++ ; 
              if(j<i){
                j= i; 
              }
            }
        }
        for( ; i<nums.length ; i++){
            nums[i] = 0 ;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.c
void moveZeroes(int* nums, int n) {
    int x = 0 ; 
    for(int i = 0 ; i<n ;i++){
        if(nums[i]!=0){
            int temp = nums[i] ; 
            nums[i] = nums[x] ; 
            nums[x] =temp;
           x++ ;
        }
    }
}